doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -131,21 +138,65 @@ public void createDoiRequest(CreateDoiRequest createDoiRequest, String username)
         Publication publication = fetchPublicationForUser(createDoiRequest, username);
         verifyThatPublicationHasNoPreviousDoiRequest(publication);
         DoiRequest newDoiRequestEntry = createDoiRequestEntry(createDoiRequest, username);
-        publication.setDoiRequest(newDoiRequestEntry);
-        publication.setModifiedDate(newDoiRequestEntry.getModifiedDate());
+        updatePublication(publication, newDoiRequestEntry);
         putItem(publication);
     }
 
     @Override
-    public void updateDoiRequest(UUID publicationIdentifier, DoiRequestStatus requestedStatusChange,
+    public void updateDoiRequest(UUID publicationIdentifier, ApiUpdateDoiRequest apiUpdateDoiRequest,
                                  String requestedByUsername, List<AccessRight> userAccessRights)
         throws ApiGatewayException {
+
+        authorizeChange(apiUpdateDoiRequest.getDoiRequestStatus(), userAccessRights, requestedByUsername);
+
         Publication publication = fetchPublicationByIdentifier(publicationIdentifier);
-        authorizeChange(requestedStatusChange, userAccessRights, requestedByUsername);
-        publication.updateDoiRequestStatus(requestedStatusChange);
+
+        DoiRequest updatedDoiRequest = updateDoiRequest(publication, apiUpdateDoiRequest, requestedByUsername);
+
+        updatePublication(publication, updatedDoiRequest);
         putItem(publication);
     }
 
+    private DoiRequest updateDoiRequest(Publication publication, ApiUpdateDoiRequest apiUpdateDoiRequest,
```

Comment:
I am not using the Pubiicaiton::updateDoiRequestStatus because it does not support adding a message and I do not want to add more logic to the datamodel before we have an agreement on whether and how we are going to do it.